cpp/src/io/utilities/config_utils.cpp

```diff
@@ -53,6 +62,28 @@ bool is_gds_enabled() { return is_always_enabled() or get_env_policy() == usage_
 
 bool is_kvikio_enabled() { return get_env_policy() == usage_policy::KVIKIO; }
 
+bool is_current_device_grace_hopper()
+{
+  int device_idx{};
+  CUDF_CUDA_TRY(cudaGetDevice(&device_idx));
+
+  // Cache the query result to avoid the expensive call to cudaGetDeviceProperties().
+  // Key: device index.
+  // Value: whether the device is a Grace Hopper.
+  static std::unordered_map<int, bool> memo;
+
+  if (auto search = memo.find(device_idx); search == memo.end()) {
+    cudaDeviceProp property{};
+    CUDF_CUDA_TRY(cudaGetDeviceProperties(&property, device_idx));
+    auto* substr_exists = std::strstr(property.name, "GH200");
```

Comment:
we probably want to treat GH100 in the same way. Should we instead use `pageableMemoryAccessUsesHostPageTables()` here? If it's a reliable way to identify a GH(like) system, it should be more future-proof. I'm just not sure if it does exactly what we need.